### PR TITLE
Code cleaning after classification refactor 1/n

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Cleaning after the classification refactor.
This PR cleans metrics that were directly removed, move to other domain or included in some other metric.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: src/torchmetrics/__init__.py [hunk 1]
```diff
@@ -21,26 +21,19 @@
...
     ConfusionMatrix,
-    CoverageError,
     Dice,
     F1Score,
     FBetaScore,
     HammingDistance,
     HingeLoss,
     JaccardIndex,
-    LabelRankingAveragePrecision,
-    LabelRankingLoss,
```
Comment (Borda) [inline src/torchmetrics/__init__.py:43]: do we have replacements for these?

File: src/torchmetrics/__init__.py [hunk 2]
```diff
@@ -21,26 +21,19 @@
...
     ROC,
     Accuracy,
     AveragePrecision,
-    BinnedAveragePrecision,
-    BinnedPrecisionRecallCurve,
-    BinnedRecallAtFixedPrecision,
     CalibrationError,
     CohenKappa,
     ConfusionMatrix,
-    CoverageError,
```
Comment (Borda) [inline src/torchmetrics/__init__.py:35]: do we have replacements for this one?

File: src/torchmetrics/__init__.py [hunk 3]
```diff
@@ -21,26 +21,19 @@
     SignalNoiseRatio,
 )
 from torchmetrics.classification import (  # noqa: E402
-    AUC,
```
Comment (Borda) [inline src/torchmetrics/__init__.py:24]: do we have replacements for this one?

File: src/torchmetrics/__init__.py [hunk 4]
```diff
@@ -21,26 +21,19 @@
...
     AUROC,
     ROC,
     Accuracy,
     AveragePrecision,
-    BinnedAveragePrecision,
-    BinnedPrecisionRecallCurve,
-    BinnedRecallAtFixedPrecision,
```
Comment (Borda) [inline src/torchmetrics/__init__.py:31]: not sure why these are removed...

File: src/torchmetrics/__init__.py (hunk 1 above)
Comment (SkafteNicki) [reply src/torchmetrics/__init__.py:43]: yes, they have just been renamed to streamline:
https://github.com/Lightning-AI/metrics/blob/fbfe974e44b157819bbe5421494e616bf63c6e6f/src/torchmetrics/classification/__init__.py#L99-L100

File: src/torchmetrics/__init__.py (hunk 2 above)
Comment (SkafteNicki) [reply src/torchmetrics/__init__.py:35]: https://github.com/Lightning-AI/metrics/blob/fbfe974e44b157819bbe5421494e616bf63c6e6f/src/torchmetrics/classification/__init__.py#L98

File: src/torchmetrics/__init__.py (hunk 3 above)
Comment (SkafteNicki) [reply src/torchmetrics/__init__.py:24]: only functional are available now and it has been moved to utilities as it is not really a metric:
https://github.com/Lightning-AI/metrics/blob/fbfe974e44b157819bbe5421494e616bf63c6e6f/src/torchmetrics/utilities/compute.py#L103-L115

File: src/torchmetrics/__init__.py (hunk 4 above)
Comment (SkafteNicki) [reply src/torchmetrics/__init__.py:31]: `AveragePrecision`, `PrecisionRecallCurve`, `RecallAtFixedPrecision` all support both binned and non-binned calculations and these specialized versions are therefore redundant

File: CHANGELOG.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Removed
 
--
+- Removed deprecated classification code ([#1251](https://github.com/Lightning-AI/metrics/pull/1251))
```
Comment (Borda) [inline CHANGELOG.md:29]: I see so let's make an extensive changelog to clearly mark what is the replacement of each removed class...

**Review by Borda — CHANGES_REQUESTED** [review]

File: src/torchmetrics/__init__.py (hunk 3 above)
Comment (Borda) [reply src/torchmetrics/__init__.py:24]: good, let's do https://github.com/Lightning-AI/metrics/pull/1251#discussion_r992214449

File: src/torchmetrics/__init__.py (hunk 4 above)
Comment (Borda) [reply src/torchmetrics/__init__.py:31]: good, let's do https://github.com/Lightning-AI/metrics/pull/1251#discussion_r992214449

File: src/torchmetrics/__init__.py (hunk 1 above)
Comment (Borda) [reply src/torchmetrics/__init__.py:43]: good, let's do https://github.com/Lightning-AI/metrics/pull/1251#discussion_r992214449

File: src/torchmetrics/__init__.py (hunk 2 above)
Comment (Borda) [reply src/torchmetrics/__init__.py:35]: good, let's do https://github.com/Lightning-AI/metrics/pull/1251#discussion_r992214449

File: CHANGELOG.md (hunk 5 above)
Comment (SkafteNicki) [reply CHANGELOG.md]: should be better now, please take a look :)

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,8 +26,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Removed
 
--
-
+- Removed deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `RecallAtFixedPrecision` ([#1163])(https://github.com/Lightning-AI/metrics/pull/1251))
```
Comment (Borda) [inline CHANGELOG.md:29]:
```suggestion
- Removed deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `RecallAtFixedPrecision` ([#1163](https://github.com/Lightning-AI/metrics/pull/1251))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -71,6 +72,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improved performance of retrieval metrics ([#1242](https://github.com/Lightning-AI/metrics/pull/1242))
 - Changed `SSIM` and `MSSSIM` update to be online to reduce memory usage ([#1231](https://github.com/Lightning-AI/metrics/pull/1231))
 
+### Deprecated
+
+- Deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `BinnedRecallAtFixedPrecision` ([#1163])(https://github.com/Lightning-AI/metrics/pull/1163))
```
Comment (Borda) [inline CHANGELOG.md:77]:
```suggestion
- Deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `BinnedRecallAtFixedPrecision` ([#1163](https://github.com/Lightning-AI/metrics/pull/1163))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -26,8 +26,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Removed
 
--
-
+- Removed deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `RecallAtFixedPrecision` ([#1163](https://github.com/Lightning-AI/metrics/pull/1251))
+- Removed deprecated `LabelRankingAveragePrecision`, `LabelRankingLoss` and `CoverageError` ([#1167](https://github.com/Lightning-AI/metrics/pull/1251))
+- Removed deprecated `KLDivergence` and `AUC` ([#1189](https://github.com/Lightning-AI/metrics/pull/1251))
```
Comment (Borda) [inline CHANGELOG.md:31]:
```suggestion
- Removed deprecated `BinnedAveragePrecision`, `BinnedPrecisionRecallCurve`, `RecallAtFixedPrecision` ([#1251](https://github.com/Lightning-AI/metrics/pull/1251))
- Removed deprecated `LabelRankingAveragePrecision`, `LabelRankingLoss` and `CoverageError` ([#1251](https://github.com/Lightning-AI/metrics/pull/1251))
- Removed deprecated `KLDivergence` and `AUC` ([#1251](https://github.com/Lightning-AI/metrics/pull/1251))
```

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (Borda) [reply CHANGELOG.md]: :+1:

**Comment by Borda** [comment]
not sure why make docs complains about types :(

**Pull request merged**